shared/constants/chat.js

```diff
@@ -8,14 +8,17 @@ import type {NoErrorTypedAction} from './types/flux'
 
 export type MessageType = 'Text'
 export type FollowState = 'You' | 'Following' | 'Broken' | 'NotFollowing'
+export type MessageState = 'pending' | 'sent' | 'failed'
 
 export type Message = {
   type: 'Text',
   message: HiddenString,
   author: string,
   timestamp: number,
-  messageID: number,
+  messageID?: number,
   followState: FollowState,
+  messageState: MessageState,
+  outboxID?: ?string,
```

Comment:
That is currently true, but I'm having trouble figuring out how to write that flow rule.  Want to have a go at it?  :)